packages/x-components/src/x-modules/queries-preview/components/query-preview.vue

```diff
@@ -222,21 +222,28 @@
           }
         }
       );
-      this.emitQueryPreviewRequestUpdated(this.queryPreviewRequest);
+      const previewItemQuery = this.queryPreviewInfo.query;
+      const isSavedQuery = !!this.previewResults[previewItemQuery];
+
+      // If the query has been saved it will emit load instead of the emitting the updated request.
+      if (isSavedQuery && this.persistInCache) {
```

Comment:
You're still not checking the status of the queryPreview request to decide if you should use the cache or not